var/spack/repos/builtin/packages/py-pyarrow/package.py

```diff
@@ -62,6 +64,8 @@ class PyPyarrow(PythonPackage, CudaPackage):
 
     depends_on("py-numpy@1.16.6:", type=("build", "run"), when="@3:")
     depends_on("py-numpy@1.14:", type=("build", "run"), when="@0.15:")
+    # Oldest supported numpy prior to Python 3.9 is 1.25
+    depends_on("py-numpy@:1.25", type=("build", "run"), when="^python@3.8")
```

Comment:
Version ranges are inclusive so this would mean could use `py-numpy` up to `1.25.*` if depending on `python@3.8`.

From 16.1.0's pyproject.toml:
```
..
    # Starting with NumPy 1.25, NumPy is (by default) as far back compatible
    # as oldest-support-numpy was (customizable with a NPY_TARGET_VERSION
    # define).  For older Python versions (where NumPy 1.25 is not yet avaiable)
    # continue using oldest-support-numpy.
    "oldest-supported-numpy>=0.14; python_version<'3.9'",
    "numpy>=1.25; python_version>='3.9'",
..
```

Reply:
I see `py-numpy` doesn't have versions older than `1.17.3` ATM.

Reply:
Am I misreading the `numpy` settings?

Reply:
Looking back at this, it may that this should be ```
depends_on("py-numpy@0.14:", when="^python@:3.8")
depends_on("py-numpy@0.25:", when="^python@3.9:")
```

Reply:
> Looking back at this, it may that this should be ``` depends_on("py-numpy@0.14:", when="^python@:3.8") depends_on("py-numpy@0.25:", when="^python@3.9:")

Or perhaps:

```
depends_on("py-numpy@0.14:1.24", when="^python@:3.8")
depends_on("py-numpy@1.25:", when="^python@3.9:")
```